src/tasks/daily/daily_battle_mixin.py

```
@@ -399,41 +431,78 @@ def _gather_retry_navigate(self, stage_name, category_name):
+
+        while left_ticket >= stages_cost[category_name] or cnt_extra_run < nums_extra_run:
             if enter_bool:
-                self.wait_click_ocr(match=re.compile("重新挑战"), box=self.box.bottom_left, log=True, time_out=5,
-                                    after_sleep=2, recheck_time=1)
+                self.to_restart(is_extra_mode=is_extra_mode, enter_box=enter_box)
             else:
-                # 根据 enter_str 决定点击位置, 如果是「挑战」, 并且刷取的是「能量淤积点」, 则点击右下四分之一区域, 避免误触
-                enter_box = self.box.bottom_right_quarter if enter_str == "挑战" and stage_name == "能量淤积点" else self.box.bottom_right
                 self.wait_click_ocr(match=re.compile(enter_str), time_out=10, after_sleep=2, box=enter_box,
                                     log=True, recheck_time=1)
+                # 如果是放弃领奖，需要点击确认
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)